cpp/arcticdb/entity/type_utils.cpp

```diff
@@ -39,32 +39,56 @@ namespace arcticdb {
         return false;
     }
 
-    std::optional<entity::TypeDescriptor> has_valid_type_promotion(
+    constexpr bool is_mixed_float_and_integer(entity::DataType left, entity::DataType right) {
+        return (is_integer_type(left) && is_floating_point_type(right)) || (is_floating_point_type(left) && is_integer_type(right));
+    }
+
+    std::optional<entity::TypeDescriptor> common_type_float_integer(const entity::TypeDescriptor& left, const entity::TypeDescriptor& right) {
```

Comment:
nit: If used only in this file mark it as static
```suggestion
    static std::optional<entity::TypeDescriptor> common_type_float_integer(const entity::TypeDescriptor& left, const entity::TypeDescriptor& right) {
```